scripts/tests/unit/test_schema_loader.py

```diff
@@ -646,6 +646,167 @@ def test_merge_non_array_attributes(self):
         }
         self.assertEqual(merged_fields, expected_fields)
 
+    def test_merge_multi_fields(self):
+        schema1 = {
+            'base': {
+                'field_details': {
+                    'multi_fields': [
```

Comment:
Your test cases should only add multi-fields to leaf fields, not to field sets. So if you need two test cases, don't hesitate to add bogus fields.

Here the `message` test cases is fine.

For the other test case, where you merge multi-fields "text" + "caseless" with "text" + "almost_text", I'd recommend two things:

* Have these MFs on a bogus field instead of `base`. Perhaps just `overwrite_field`, doesn't need to be a real field name.
* Tweak the overridden `text` multi-field to demonstrate that it's ripped out and replaced by the second; it's not merging the two MF definitions together.

Original MF:

```
            'overwrite_field': {
                'field_details': {
                    'multi_fields': [
                        {
                            'type': 'text',
                            'name': 'text',
                            'norms': true
                        }, ...
```

Custom MF:

```
            'overwrite_field': {
                'field_details': {
                    'multi_fields': [
                        {
                            'type': 'text',
                            'name': 'text'
                        }, ...
```

Resulting `text` MF:

```
                        {
                            'type': 'text',
                            'name': 'text'
                        }
```

In other words, the MF themselves aren't merged together.

Reply:
👍  done.